### PR TITLE
Introduce new block syntax

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request  
Part of https://github.com/pimcore/pimcore/pull/17664

### CODE REVIEW

File: doc/03_Documents/01_Editables/06_Block.md [hunk 1]
```diff
@@ -9,13 +9,13 @@ The items in the loop as well as their order can be defined by the editor with t
...
+| `limit`     | integer   | Max. amount of iterations.                                                                                                                                                                                              |
+| `reload`    | bool      | Reload editmode on add, move or remove (default=false)                                                                                                                                                                  |
+| `default`   | integer   | If block is empty, this specifies the iterations at startup.                                                                                                                                                            |
+| `manual`    | bool      | Forces the manual mode, which enables a complete custom HTML implementation for blocks, for example using `<table>` elements <br/> <b>Deprecated</b> Will be removed in Pimcore 12 use `endpimcoremanualblock` instate. |
```
Comment (jdreesen) [inline doc/03_Documents/01_Editables/06_Block.md]:
```suggestion
| `manual`    | bool      | Forces the manual mode, which enables a complete custom HTML implementation for blocks, for example using `<table>` elements <br/> <b>Deprecated</b> Will be removed in Pimcore 12 use `endpimcoremanualblock` instead. |
```

File: doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 ## Pimcore 11.5.0
 ### [Events]
 - `context` property of `ResolveUploadTargetEvent` is deprecated. Use `setArgument()` method instead.
+- `pimcore_block` Twig extension is deprecated. Use `pimcoreblock` or `pimcoremanualblock`instead.
```
Comment (jdreesen) [inline doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- `pimcore_block` Twig extension is deprecated. Use `pimcoreblock` or `pimcoremanualblock` instead.
```

File: doc/03_Documents/01_Editables/06_Block.md [hunk 3]
```diff
@@ -9,13 +9,13 @@ The items in the loop as well as their order can be defined by the editor with t
...
+| `limit`     | integer   | Max. amount of iterations.                                                                                                                                                                                              |
+| `reload`    | bool      | Reload editmode on add, move or remove (default=false)                                                                                                                                                                  |
+| `default`   | integer   | If block is empty, this specifies the iterations at startup.                                                                                                                                                            |
+| `manual`    | bool      | Forces the manual mode, which enables a complete custom HTML implementation for blocks, for example using `<table>` elements <br/> <b>Deprecated</b> Will be removed in Pimcore 12 use `endpimcoremanualblock` instead. |
```
Comment (markus-moser) [inline doc/03_Documents/01_Editables/06_Block.md]: Should we change `endpimcoremanualblock` to `pimcoremanualblock` here?

File: doc/03_Documents/01_Editables/06_Block.md [hunk 4]
```diff
@@ -70,9 +89,28 @@ And in the frontend of the application:
...
+            {% do _block.blockStart() %}
+                <div style="width:200px; height:200px;border:1px solid black;">
+                    {{ pimcore_input("myInput") }}
+                </div>
+            {% do _block.blockEnd() %}
+        </td>
+     {% endblockiterate %}
+     </tr>
+</table>
+{% endpimcoreblock %}
```
Comment (markus-moser) [inline doc/03_Documents/01_Editables/06_Block.md]: Shouldn't this be `endpimcoremanualblock`?

**Review by markus-moser — CHANGES_REQUESTED** [review]
Please check the 2 comments in the docs, the rest LGTM :)

File: lib/Twig/Node/ManuelBlockNode.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+namespace Pimcore\Twig\Node;
+
+use Pimcore\Twig\Options\BlockOptions;
+use Twig\Compiler;
+use Twig\Node\Node;
+
+/**
+ * @internal
+ */
+final class ManuelBlockNode extends Node
```
Comment (NiklasBr) [inline lib/Twig/Node/ManuelBlockNode.php]:
```suggestion
final class ManualBlockNode extends Node
```

Probably need to rename file also.

File: lib/Twig/Options/BlockOptions.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    private function getReloadAsString(): string
+    {
+        if ($this->reload) {
+            return 'true';
+        }
+
+        return 'false';
+    }
+
+    private function getManuelAsString(): string
```
Comment (NiklasBr) [inline lib/Twig/Options/BlockOptions.php]:
```suggestion
    private function getManualAsString(): string
```

File: lib/Twig/Options/BlockOptions.php [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+            return 'true';
+        }
+
+        return 'false';
+    }
+
+    public function toString(): string
+    {
+        $options = '[';
+        $options .= "'manual' => " . $this->getManuelAsString() .',';
```
Comment (NiklasBr) [inline lib/Twig/Options/BlockOptions.php]:
```suggestion
        $options .= "'manual' => " . $this->getManualAsString() .',';
```

File: lib/Twig/TokenParser/ManualBlockParser.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Twig\TokenParser;
+
+use Pimcore\Twig\Node\ManuelBlockNode;
```
Comment (NiklasBr) [inline lib/Twig/TokenParser/ManualBlockParser.php]:
```suggestion
use Pimcore\Twig\Node\ManualBlockNode;
```

File: lib/Twig/TokenParser/ManualBlockParser.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        $stream->expect(Token::BLOCK_END_TYPE);
+
+        $bodyNode = $this->parser->subparse([$this, 'decideIterateEnd'], true);
+        $stream->expect(Token::BLOCK_END_TYPE);
+
+        $endNode = $this->parser->subparse([$this, 'decidePimcoreManualBlockEnd'], true);
+
+        $this->parser->getStream()->expect(Token::BLOCK_END_TYPE);
+
+        return new ManuelBlockNode(
```
Comment (NiklasBr) [inline lib/Twig/TokenParser/ManualBlockParser.php]:
```suggestion
        return new ManualBlockNode(
```

**Review by NiklasBr — COMMENTED** [review]
I suppose this is just a language quirk, but Manuel -> Manual

**Comment by martineiber** [comment]
> I suppose this is just a language quirk, but Manuel -> Manual

Good catch. Thank you :smiley:

**Review by markus-moser — APPROVED** [review]

**Pull request merged**